app/webpacker/controllers/login_modal_controller.js

```diff
@@ -1,8 +1,74 @@
 import { Controller } from "stimulus"
+import CableReady from "cable_ready"
 
 export default class extends Controller {
+  static targets = ["background", "modal", "email"]
+  static values = { email: String }
+
+  connect() {
+    if(this.hasModalTarget) {
+      window.addEventListener("login:modal:open", this.open)
+
+      if(location.hash.substr(1).includes("/login")) {
+        this.open()
+      }
+    }
+  }
+
   call(event) {
     event.preventDefault()
     window.dispatchEvent(new Event("login:modal:open"))
   }
+
+  emailOnInput(event) {
+    this.emailValue = event.currentTarget.value
+    this.emailTargets.forEach((element) => {
+      element.value = this.emailValue
+    })
+  }
+
+  open = () => {
```

Comment:
Any specific on using this syntax for this one?

Reply:
Yeah. It's a bit complex but I'll have a go at describing it...

So it's related to the way `this` works in Javascript in relation to binding functions to listeners. Depending on how it's done, the `this` context can be quite different. At the top of this class I'm binding `this.open()` to a custom DOM Event, so that other controllers can call the modal to open even if they're not in the same scope (like if they're in a totally different part of the DOM tree). And that's done with:
```js
window.addEventListener("login:modal:open", this.open)
```
So... if I _wasn't_ using an arrow-function here, I wouldn't be able to access the controller itself from within that function when it was called as part of the event being triggered. Does that makes sense? Essentially it's ensuring the controller itself is the context (`this`), so things like `this.modalTarget` are still accessible.

If I wasn't using the arrow-function syntax here, the code inside the `open()` function would have the function _itself_ as the `this` context, and wouldn't be able to call things like `this.modalTarget` (which are defined outside of that function). I would have to figure out some other way of binding the controller correctly to `this` inside that function when it's used as a callback, and it might look a bit like this:
```js
connect() {
  window.addEventListener("login:modal:open", this.open.bind(this))
}

open() {
  // Now I can access this.modalTarget here because "this" is now the controller, not the open() function.
  // Without that bind() call, this.modalTarget would be undefined here (not usable/accessible) :/
}

disconnect() {
  window.removeEventListener("login:modal:open", this.open.bind(this))
}
```
or I could assign the "bound" version of the callback to a variable, like this:
```js
connect() {
  this.boundOpenFunction = this.open.bind(this)
  window.addEventListener("login:modal:open", this.boundOpenFunction)
}

open() {
  // Same as above :P
}

disconnect() {
  window.removeEventListener("login:modal:open", this.boundOpenFunction)
}
```
but both of these options are a bit messy, and add a lot of extra mental overhead.

Instead, by just switching the definition of the `open()` function an arrow-function we can just avoid the need to jump through those extra hoops with bindings, and it works as expected when called directly or when used as a callback. 
```diff
-- open() {
++ open = () => {
```

:ok_hand:

Reply:
Yes, all clear! (I knew about the `this` and arrow functions, but I didn't see clearly you need to have this binded to the controller with the event listener, sorry and thanks for the *very* clear explanation!)